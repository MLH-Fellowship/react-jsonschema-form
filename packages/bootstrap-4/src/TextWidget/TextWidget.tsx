import React from "react";

import Form from "react-bootstrap/Form";

import { WidgetProps } from "@rjsf/core";

export interface TextWidgetProps extends WidgetProps {
  type?: string;
}

const TextWidget = ({
  id,
  required,
  readonly,
  disabled,
  type,
  label,
  value,
  onChange,
  onBlur,
  onFocus,
  autofocus,
  options,
  schema,
  rawErrors = [],
  formContext,
  ...textFieldProps
}: TextWidgetProps) => {
  const _onChange = ({
    target: { value },
  }: React.ChangeEvent<HTMLInputElement>) =>
    onChange(value === "" ? options.emptyValue : value);
  const _onBlur = ({ target: { value } }: React.FocusEvent<HTMLInputElement>) =>
    onBlur(id, value);
  const _onFocus = ({
    target: { value },
  }: React.FocusEvent<HTMLInputElement>) => onFocus(id, value);

  return (
    <Form.Group controlId={id} className="mb-0">
      <Form.Label>
        {label || schema.title}
        {required ? "*" : null}
      </Form.Label>
      <Form.Control
        id={id}
        autoFocus={autofocus}
        required={required}
        disabled={disabled || readonly}
        name={name}
        type={type || (schema.type as string)}
        value={value || value === 0 ? value : ""}
        onChange={_onChange}
        onBlur={_onBlur}
        onFocus={_onFocus}
        {...textFieldProps}
      />
    </Form.Group>
  );
};

export default TextWidget;
