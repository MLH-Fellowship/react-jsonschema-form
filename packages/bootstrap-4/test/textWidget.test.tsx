import React from "react";
import TextWidget from "../src/TextWidget";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";

describe("TextWidget", () => {
  test("simple", () => {
    const schema: JSONSchema7 = {
      type: "array",
      items: {
        type: "string",
      },
    };
    const mockHandler = (): void => void 0;
    const tree = renderer
      .create(
        <TextWidget
          schema={schema}
          required
          disabled={false}
          readonly
          autofocus
          label="Some simple label"
          onChange={mockHandler}
          onBlur={mockHandler}
          onFocus={mockHandler}
          multiple={false}
          rawErrors={[""]}
          value="value"
          options={{}}
          type="text"
          formContext={{}}
          id="_id"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
