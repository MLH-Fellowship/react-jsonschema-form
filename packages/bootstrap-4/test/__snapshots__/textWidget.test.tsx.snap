// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWidget simple 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="_id"
  >
    Some simple label
    *
  </label>
  <input
    autoFocus={true}
    className="form-control"
    disabled={true}
    id="_id"
    multiple={false}
    name="nodejs"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
    value="value"
  />
</div>
`;
